### DISCUSSION TITLE
Problem getting matrixes? "$ operator not defined"?

### DISCUSSION THREAD

**Discussion by mfrenkel16** [GreenleafLab/ArchR#1886] [Questions / Documentation] [answered]
Hi,

I am trying to get the matrixes (e.g. TileMatrix, GeneScoreMatrix etc.) from an arrow file/project for custom analysis outside of ArchR. All of the analysis within ArchR works and I'm able to retrieve the matrixes with either `getMatrixFromArrow` or `getMatrixFromProject `. However, I can't find a way to access the actual count data within. For instance, the command `test3@assays$GeneScoreMatrix` results in the error "$ operator not defined for this S4 class" even though tabbing to complete after $ works suggesting that is a way to access the matrix. This error occurs regardless of which matrix I try retrieving and regardless of whether I get it from the arrow file or the project.

<img width="525" alt="Screenshot 2023-03-18 at 8 09 45 AM" src="https://user-images.githubusercontent.com/91026492/226114289-3e219e8d-7edb-4f54-b094-d08d11761625.png">

Am I going about this wrong or can I access the actual data within these matrixes in some other way? 

Thanks!
Max

**Comment by mfrenkel16** [comment, marked as answer]
Solved (of course 15 min after posting, despite >1 day of troubleshooting). 

For others in a similar boat, here's the way to access the matrix: test3@assays@data$GeneScoreMatrix (not test3@assays$GeneScoreMatrix even though tab to complete would suggest that's a way to access it)

**Reply by rcorces** [reply]
just to add to this, typically this is done using `assays(test3)`

**Reply by mfrenkel16** [reply]
Thank you! This is obviously a case of me not knowing the structure of these files well enough so I appreciate the follow up.
